provisioner/terraform/tfparse/tfextract.go

```diff
@@ -1,44 +1,103 @@
 package tfparse
 
 import (
+	"archive/zip"
+	"bytes"
 	"context"
 	"encoding/json"
+	"io"
 	"os"
 	"slices"
 	"sort"
+	"strconv"
 	"strings"
 
+	"github.com/coder/coder/v2/archive"
+	"github.com/coder/coder/v2/provisionersdk"
 	"github.com/coder/coder/v2/provisionersdk/proto"
 
 	"github.com/hashicorp/hcl/v2"
 	"github.com/hashicorp/hcl/v2/hclparse"
 	"github.com/hashicorp/hcl/v2/hclsyntax"
 	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+	"github.com/zclconf/go-cty/cty"
+	"golang.org/x/exp/maps"
 	"golang.org/x/xerrors"
 
 	"cdr.dev/slog"
 )
 
-// WorkspaceTags extracts tags from coder_workspace_tags data sources defined in module.
-func WorkspaceTags(ctx context.Context, logger slog.Logger, module *tfconfig.Module) (map[string]string, error) {
-	workspaceTags := map[string]string{}
+// NOTE: This is duplicated from coderd but we can't import it here without
+// introducing a circular dependency
+const maxFileSizeBytes = 10 * (10 << 20) // 10 MB
 
-	for _, dataResource := range module.DataResources {
+// parseHCLFiler is the actual interface of *hclparse.Parser we use
+// to parse HCL. This is extracted to an interface so we can more
+// easily swap this out for an alternative implementation later on.
+type parseHCLFiler interface {
```

Comment:
nit: funny name :)

Reply:
I know :D I'm just following the Go convention of `<function name>er`.